### PR TITLE
Android import cpufeatures in cocos.

### PR DESCRIPTION
**Pull request by perminovVS**
Add import cpufeatures internal.
Now cpufeatures imported in 3rd party libs webp.
If not use webp, android not build.

### CODE REVIEW

File: cocos/Android.mk [hunk 1]
```diff
@@ -275,10 +275,14 @@ LOCAL_STATIC_LIBRARIES += recast_static
...
 
 LOCAL_WHOLE_STATIC_LIBRARIES := cocos2dxandroid_static
+LOCAL_WHOLE_STATIC_LIBRARIES += cpufeatures
 
 # define the macro to compile through support/zip_support/ioapi.c
 LOCAL_CFLAGS   :=  -DUSE_FILE32API
 LOCAL_CFLAGS   +=  -fexceptions
+ifeq ($(TARGET_ARCH_ABI),armeabi-v7a)
+    LOCAL_CFLAGS += -DHAVE_NEON=1
+endif
```
Comment (minggo) [inline cocos/Android.mk]: Should check armeabi-v8 too?

**Comment by minggo** [comment]
> If not use webp, android not build.

I will check it.

File: cocos/Android.mk (hunk 1 above)
Comment (minggo) [inline cocos/Android.mk]: sorry, it is `arm64-v8a`

**Comment by minggo** [comment]
Yep, because `MathUtils.cpp` depends on cpu-feature, and this issue fix it.

File: cocos/Android.mk (hunk 1 above)
Comment (minggo) [inline cocos/Android.mk]: And as i know, `armeabi-v7a` may doesn't support neon, can refer to #9968 for detail information. So we can not add this line.

Comment (minggo) [inline cocos/Android.mk]: @perminovVS what's your idea about it? And i think the modification should not be part of this PR.

Comment (perminovVS) [inline cocos/Android.mk]: @minggo i get it from https://github.com/cocos2d/cocos2d-x-3rd-party-libs-bin/blob/v3/webp/prebuilt/android/Android.mk

If armeabi-v7a cpu not always support neon instruction, need check it runtime and change use code.

I see MathUtil.cpp, code:

``` c++
bool MathUtil::isNeon32Enabled()
{
#ifdef USE_NEON32
    return true;
#elif (defined (INCLUDE_NEON32) && (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID) )
    class AnrdoidNeonChecker
    {
    public:
        AnrdoidNeonChecker()
        {
            if (android_getCpuFamily() == ANDROID_CPU_FAMILY_ARM && (android_getCpuFeatures() & ANDROID_CPU_ARM_FEATURE_NEON) != 0)
                _isNeonEnabled = true;
            else
                _isNeonEnabled = false;
        }
        bool isNeonEnabled() const { return _isNeonEnabled; }
    private:
        bool _isNeonEnabled;
    };
    static AnrdoidNeonChecker checker;
    return checker.isNeonEnabled();
#else
    return false;
#endif
}
```

It now implemented, additionally do not need to do anything.
For arm64-v8a not need define HAVE_NEON

``` c++
#elif (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID)
    #if defined (__arm64__) || defined (__aarch64__)
    #define USE_NEON64
    #define INCLUDE_NEON64
    #elif defined (__ARM_NEON__)
    #define INCLUDE_NEON32
    #else
    #endif
#else
```

Need add log message, for mobile platform, support neon.

Comment (minggo) [inline cocos/Android.mk]: Will you please add a new PR for it, we can discuss there. I will merge the PR if you remove the codes.

**Comment by perminovVS** [comment]
[ci rebuild]
